lighthouse-core/test/scripts/run-mocha-tests.js

```diff
@@ -299,6 +300,11 @@ async function runMocha(tests, mochaArgs, invocationNumber) {
   const rootHooksPath = mochaArgs.require || '../test-env/mocha-setup.js';
   const {rootHooks} = await import(rootHooksPath);
 
+  let mocksFilePath;
+  if (tests.length === 1) {
```

Comment:
Leaning on the definitions of "test isolation" above to indirectly control when these `mocks` files are called. When isolation stuff is dropped (I'm working on it now), the runner can be rewritten slightly to instead run files by themselves if they have a `mocks` file defined. It wouldn't be a big change to do that in this PR but, I didn't think it was necessary, preferring a more incremental change.